src/Mailgun/Api/MailingList.php

```diff
@@ -0,0 +1,307 @@
+        Assert::stringNotEmpty($address);
+        Assert::nullOrStringNotEmpty($name);
+        Assert::nullOrStringNotEmpty($description);
+        Assert::oneOf($accessLevel, ['readonly', 'members', 'everyone']);
+
+        $params = [
+            'address' => $address,
+            'name' => $name,
+            'description' => $description,
+            'access_level' => $accessLevel,
+        ];
+
+        $response = $this->httpPost('/v3/lists', $params);
+
+        return $this->hydrateResponse($response, CreateResponse::class);
+    }
+
+    /**
+     * Returns a single mailing list.
+     *
+     * @param string $address Address of the mailing list
+     *
+     * @return ShowResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function show($address)
+    {
+        Assert::stringNotEmpty($address);
+
+        $response = $this->httpGet(sprintf('/v3/lists/%s', $address));
+
+        return $this->hydrateResponse($response, ShowResponse::class);
+    }
+
+    /**
+     * Updates a mailing list.
+     *
+     * @param string $address    Address of the mailing list
+     * @param array  $parameters Array of field => value pairs to update
+     *
+     * @return UpdateResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function update($address, $parameters = [])
+    {
+        Assert::stringNotEmpty($address);
+        Assert::isArray($parameters);
+
+        foreach ($parameters as $field => $value) {
+            switch ($field) {
+                case 'address':
+                case 'name':
+                case 'description':
+                    Assert::stringNotEmpty($value);
+
+                    break;
+                case 'access_level':
+                    Assert::oneOf($value, ['readonly', 'members', 'everyone']);
+
+                    break;
+                default:
+                    unset($parameters[$field]);
+
+                    break;
+            }
+        }
+
+        $response = $this->httpPut(sprintf('/v3/lists/%s', $address), $parameters);
+
+        return $this->hydrateResponse($response, UpdateResponse::class);
+    }
+
+    /**
+     * Removes a mailing list from the domain.
+     *
+     * @param string $address Address of the mailing list
+     *
+     * @return DeleteResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function delete($address)
+    {
+        Assert::stringNotEmpty($address);
+
+        $response = $this->httpDelete(sprintf('/v3/lists/%s', $address));
+
+        return $this->hydrateResponse($response, DeleteResponse::class);
+    }
+
+    /**
+     * Returns a paginated list of members of the mailing list.
+     *
+     * @param string $address    Address of the mailing list
+     * @param int    $limit      Maximum number of records to return (optional: 100 by default)
+     * @param mixed  $subscribed `yes` to lists subscribed, `no` for unsubscribed. list all if null
```

Comment:
`mixed` type should be updated to match with the expected type on the PHP Docs. In this case, we could set `string|null` due the fact that we expect a string or a non-defined value.